dsp_permissions_scripts/doap/doap_get.py

```diff
@@ -28,17 +29,20 @@ def _get_all_doaps_of_project(project_iri: str, dsp_client: DspClient) -> list[D
 def create_doap_from_admin_route_response(permission: dict[str, Any]) -> Doap:
     """Deserializes a DOAP from JSON as returned by /admin/permissions/doap/{project_iri}"""
     scope = create_scope_from_admin_route_object(permission["hasPermissions"])
-    doap = Doap(
-        target=DoapTarget(
+    target: GroupDoapTarget | EntityDoapTarget
+    if permission.get("forGroup"):
+        target = GroupDoapTarget(project_iri=permission["forProject"], group=Group(val=permission["forGroup"]))
+    else:
```

Comment:
Since this PR is "cosmetic" anyways: I think this if-clause might be a nice case for structural decomposition of the dict using pattern matching.  
Something along the lines of
```python
match permissions:
    case {"forProject": project_iri, "forGroup": group}:
        target = GroupDoapTarget(project_iri, group)
    case {"forProject": project_iri, **p}:
        target = EntityDoapTarget(
            project_iri=project_iri,
            resource_class=p.get("forResourceClass"),
            property=p.get("forProperty")
        )
```
though you'd have to try it out if it actually works as hoped - I'm not using python pattern matching enough these days, to be really confident

Reply:
Thanks for the idea. With small modifications, your code works.